### PR TITLE
Fi 1072 add allergyintolerance reaction and manifestation

### PR DESCRIPTION
**Pull request by czh-orz**
AllergyIntolerance.reaction.manifestation is now must support in 3.1.1. This PR updates the test set to include this element.
image
image

### CODE REVIEW

**Comment by arscan** [comment]
Was there any opportunity to fill data in here with something 'realistic' or is that too much of a lift?

**Comment by czh-orz** [comment]
> Was there any opportunity to fill data in here with something 'realistic' or is that too much of a lift?

Yep, just updated to "skin rash" instead.

**Comment by arscan** [comment]
Ideally we want to keep these hardcoded changes to a minimum.  So could you just put it in one resource instead of all?  Ideally, this would check to make sure that synthea didn't already fill in these (I know the current version doesn't, but i bet future versions will), and only add one (and only one) if none have been added already.

**Comment by czh-orz** [comment]
> Ideally we want to keep these hardcoded changes to a minimum. So could you just put it in one resource instead of all? Ideally, this would check to make sure that synthea didn't already fill in these (I know the current version doesn't, but i bet future versions will), and only add one (and only one) if none have been added already.
Updated to only make changes if no allergy

Updated to only make changes if no allergyintolerance has a reaction.manifestation and to only change one allergyintolerance.

File: lib/modifications.rb [hunk 1]
```diff
@@ -71,6 +71,29 @@ def self.modify!(results, random_seed = 3)
         end
       end
 
+      # add reaction.manifestation to allergy intolerance result because of must support changes in 3.1.1
+      already_contains_reaction_manifestation = false
```
Comment (Jammjammjamm) [inline lib/modifications.rb:75]: `.any?` will handle the early exit from the iteration.
```
already_contains_reaction_manifestation = 
  results.any? do |bundle|
    bundle
      .entry
      .select { |e| e.resource.is_a? FHIR::AllergyIntolerance }
      .map(&:resource)
      .any? do |resource|
        ...
      end
  end
```

**Review by Jammjammjamm — APPROVED** [review]

File: lib/modifications.rb (hunk 1 above)
Comment (czh-orz) [reply lib/modifications.rb]: Nice, definitely cleaner. Updated to these changes.

**Review by okeefm — APPROVED** [review]
Looks good to me.

**Pull request merged**